packetbeat/sniffer/afpacket_linux.go

```diff
@@ -40,6 +43,85 @@ type afpacketHandle struct {
 	promiscPreviousStateDetected bool
 	device                       string
 	log                          *logp.Logger
+	metrics                      *metrics
+}
+
+type metrics struct {
```

Comment:
I'd prefer that the metrics code go below the afpacket code as it's secondary to the function of the code here.